native/src/widget/operation/scrollable.rs

```diff
@@ -34,7 +37,52 @@ pub fn snap_to<T>(target: Id, offset: RelativeOffset) -> impl Operation<T> {
     SnapTo { target, offset }
 }
 
-/// The amount of offset in each direction of a [`Scrollable`].
+/// Produces an [`Operation`] that scrolls the widget with the given [`Id`] to
+/// the provided [`AbsoluteOffset`].
+pub fn scroll_to<T>(target: Id, offset: AbsoluteOffset) -> impl Operation<T> {
+    struct ScrollTo {
+        target: Id,
+        offset: AbsoluteOffset,
+    }
+
+    impl<T> Operation<T> for ScrollTo {
+        fn container(
+            &mut self,
+            _id: Option<&Id>,
+            operate_on_children: &mut dyn FnMut(&mut dyn Operation<T>),
+        ) {
+            operate_on_children(self)
+        }
+
+        fn scrollable(&mut self, state: &mut dyn Scrollable, id: Option<&Id>) {
+            if Some(&self.target) == id {
+                state.scroll_to(self.offset);
+            }
+        }
+    }
+
+    ScrollTo { target, offset }
+}
+
+/// The current absolute & relative offset of a [`Scrollable`]
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub struct CurrentOffset {
+    /// The [`AbsoluteOffset`] of a [`Scrollable`]
+    pub absolute: AbsoluteOffset,
+    /// The [`RelativeOffset`] of a [`Scrollable`]
+    pub relative: RelativeOffset,
+}
```

Comment:
There's some redundancy here.

I think it'd be better to have a `Window` or `Viewport` type with the already existing private `Offset`, the content bounds, and the scrollable bounds.

Then we can expose `relative_offset` and `absolute_offset` methods in this type that compute the actual values from a single source of truth.

Reply:
Great suggestion  6ad5e03d]